### PR TITLE
Update web.xml to require login for viewing lectures

### PR DESCRIPTION
**Pull request by n-wach**
This PR:
* Ensures that anyone viewing the HTML is logged in

Here is a video demonstrating how it works: https://drive.google.com/file/d/1P6FQ-XC67h8twJNFGvsP9FC9b2AA4rJQ/view?usp=sharing

### CODE REVIEW

**Comment by n-wach** [comment]
This also adds an index.html stub

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      String login = userService.createLoginURL(requestUrl);
+      response.sendRedirect(login);
+      return;
+    }
+
+    sendFile(response, pageFile);
+  }
+
+  /**
+   * Reads a file from disk and sends to the {@code response} output stream. This also
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java:63]: nit: should the additional detail be put as a separate paragraph?

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      // Set to binary type if MIME mapping not found.
+      mimeType = "application/octet-stream;";
+    }
+    response.setContentType(mimeType);
+    response.setContentLength((int) file.length());
+
+    FileInputStream inStream = new FileInputStream(file);
+    OutputStream outStream = response.getOutputStream();
+
+    byte[] buffer = new byte[4096];
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java:81]: What is the significance of 4096?

File: src/test/java/com/googleinterns/zoomtube/servlets/pages/PageServletTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  @Test(expected = FileNotFoundException.class)
+  public void doGet_loggedIn_expectFileNotFound() throws Exception {
+    testServices.setEnvIsLoggedIn(true);
+    servlet = new PageServlet("does_not_exist.html");
+    servlet.init();
+
+    servlet.doGet(request, response);
+
+    // This test does not run in the correct directory to reference existing HTML
```
Comment (amuamushu) [inline src/test/java/com/googleinterns/zoomtube/servlets/pages/PageServletTest.java:50]: nit: can you move the comment to the top of the code block for this method?

Comment (n-wach) [reply src/test/java/com/googleinterns/zoomtube/servlets/pages/PageServletTest.java:50]: Fixed!

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java (hunk 2 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java:81]: added BUFFER_SIZE constant

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java (hunk 1 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java:63]: Done!

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/ListServlet.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+package com.googleinterns.zoomtube.servlets.pages;
+
+import javax.servlet.annotation.WebServlet;
+
+/**
+ * Serves the lecture-list page to logged in users.
+ */
+@WebServlet("/list/*")
+public class ListServlet extends PageServlet {
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/servlets/pages/ListServlet.java:23]: Should this class be renamed LectureListServlet if it serves the lecture list?

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/ListServlet.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  public ListServlet() {
+    super("lecture-list.html");
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/servlets/pages/ListServlet.java:25]: If the only thing you are passing is a string to the constructor why do you need to extend the class?

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Serves a static file for logged in users.  If a user isn't logged in, they are
+ * redirected to a login page.
+ */
+public class PageServlet extends HttpServlet {
+  private UserService userService;
+  private String pageFile;
+  private int BUFFER_SIZE = 4096;
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java:39]: Seems this should be private static final

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Serves a static file for logged in users.  If a user isn't logged in, they are
+ * redirected to a login page.
+ */
+public class PageServlet extends HttpServlet {
+  private UserService userService;
+  private String pageFile;
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java:38]: seems this can be private final

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/ListServlet.java (hunk 5 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/pages/ListServlet.java:25]: I need the class declaration for the @WebServlet annotation to create the servlet

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java (hunk 6 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java:39]: Fixed!

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/ListServlet.java (hunk 4 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/pages/ListServlet.java]: Changed to LectureListPageServlet

File: src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java (hunk 7 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/pages/PageServlet.java]: Done!

**Comment by n-wach** [comment]
Also simplified HTML file names as they must now be included in url

**Comment by f-camp** [comment]
This looks great Nathan! If you can add a video of how the urls work now that would be great. Both for the non logged in state and the logged in state.

**Comment by n-wach** [comment]
@f-camp Uploaded here: https://drive.google.com/file/d/1P6FQ-XC67h8twJNFGvsP9FC9b2AA4rJQ/view

File: src/main/webapp/list.html [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<meta charset="utf-8" />
```
Comment (billie-cat) [inline src/main/webapp/list.html:2]: nit: this file name is too vague. List does not really say much about what the file is. I would rename is lectures.html or lecture-list.html

File: src/main/webapp/index.html [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE html>
```
Comment (billie-cat) [inline src/main/webapp/index.html:1]: nit: index.html is a vague name, how about lecture-index.html

File: src/main/webapp/view.html [hunk 10]
```diff
@@ -2,9 +2,9 @@
 <meta charset="utf-8" />
```
Comment (billie-cat) [inline src/main/webapp/view.html:2]: why did you rename it to view.html? lecture-view.html seemed like a more descriptive name.

Comment (n-wach) [reply src/main/webapp/view.html:2]: File name present in URL--we want to keep them short

File: src/main/webapp/list.html (hunk 8 above)
Comment (n-wach) [reply src/main/webapp/list.html:2]: File name present in URL--we want to keep them short

File: src/main/webapp/index.html (hunk 9 above)
Comment (n-wach) [reply src/main/webapp/index.html:1]: index.html is a standard file name displayed when viewing a directory (in this case the root dir "/").  Can't be changed
